sql/core.go

```diff
@@ -368,20 +369,53 @@ type SystemVariableRegistry interface {
 	AssignValues(vals map[string]interface{}) error
 	// NewSessionMap returns a map of system variables values that can be used by a session
 	NewSessionMap() map[string]SystemVarValue
-	// GetGlobal returns the global value of the system variable with the given name
+	// GetGlobal returns the current global value of the system variable with the given name
 	GetGlobal(name string) (SystemVariable, interface{}, bool)
 	// SetGlobal sets the global value of the system variable with the given name
 	SetGlobal(name string, val interface{}) error
 	// GetAllGlobalVariables returns a copy of all global variable values.
 	GetAllGlobalVariables() map[string]interface{}
 }
 
-// SystemVariable represents a system variable.
-type SystemVariable struct {
+// SystemVariable is used to system variables.
+type SystemVariable interface {
+	// GetName returns the name of the sv. Case-sensitive.
+	GetName() string
+	// GetType returns the type of the sv.
+	GetType() Type
+	// GetSessionScope returns SESSION scope of the sv.
+	GetSessionScope() SystemVariableScope
+	// SetDefault sets the default value of the sv.
+	SetDefault(any)
+	// GetDefault returns the defined default value of the sv.
+	// This is used for resetting some variables to initial default/reset value.
+	GetDefault() any
+	// InitValue sets value without validation.
+	// This is used for setting the initial values internally
+	// using pre-defined variables or for test-purposes.
+	InitValue(val any, global bool) (SystemVarValue, error)
+	// SetValue sets the value of the sv of given scope, global or session
+	// It validates setting value of correct scope,
+	// converts the given value to appropriate value depending on the sv
+	// and it returns the SystemVarValue with the updated value.
+	SetValue(val any, global bool) (SystemVarValue, error)
+	// IsReadOnly checks whether the variable is read only.
+	// It returns false if variable can be set to a value.
+	IsReadOnly() bool
+	// IsGlobalOnly checks whether the scope of the variable is global only.
+	IsGlobalOnly() bool
+	// String returns the name of the sv to be displayed.
+	String(string) string
```

Comment:
I'd add the variable name to the interface, or explain what it's for in the comment. Normally, `String()` doesn't take a parameter so it should be clear why we're not following the trend here.